atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraTimestampUtils.java

```diff
@@ -96,7 +101,12 @@ public static ByteBuffer constructCheckAndSetMultipleQuery(Map<String, Pair<byte
     }
 
     public static boolean isValidTimestampData(byte[] data) {
-        return data != null && data.length == Long.BYTES;
+        try {
+            TimestampBoundStoreEntry.createFromBytes(data);
+            return true;
+        } catch (Exception e) {
```

Comment:
I agree we should log, though probably not higher than info since eventually when timelock is a thing, on startup we want this to fail for the embedded service.